dbms/src/Storages/KVStore/MultiRaft/Disagg/FastAddPeerCache.cpp

```diff
@@ -148,8 +148,23 @@ ParsedCheckpointDataHolderPtr buildParsedCheckpointData(Context & context, const
         // Note the `data_file_id` in temp ps is all lock key, we need transform them to data key before write to local ps.
         for (auto & record : records)
         {
+            {
+                auto id_storage_type = UniversalPageIdFormat::getUniversalPageIdType(record.page_id);
+                // Filter remote peer's local page storage pages.
+                if (id_storage_type == StorageType::LocalKV)
+                {
+                    continue;
+                }
+            }
             if (record.type == PS::V3::EditRecordType::VAR_ENTRY)
             {
+                if (!record.entry.checkpoint_info.data_location.isValid())
+                {
+                    throw Exception(
+                        ErrorCodes::LOGICAL_ERROR,
+                        "buildParsedCheckpointData: can't put remote page with empty data_location, page_id={}",
+                        record.page_id);
+                }
```

Comment:
replace with `RUNTIME_CHECK_MSG`?

Reply:
No, this is a recoverable exception, on which TiFlash will fallback to legacy snapshot

Reply:
```
#define RUNTIME_CHECK_MSG(condition, fmt, ...)                                                                      \
    do                                                                                                              \
    {                                                                                                               \
        if (unlikely(!(condition)))                                                                                 \
        {                                                                                                           \
            throw ::DB::Exception(                                                                                  \
                ::DB::FmtBuffer().append("Check " #condition " failed: ").fmtAppend(fmt, ##__VA_ARGS__).toString(), \
                ::DB::ErrorCodes::LOGICAL_ERROR);                                                                   \
        }                                                                                                           \
    } while (false)
```

RUNTIME_CHECK_MSG do the same thing.

Reply:
However, I prefer a throw Exception way, since I am not expecting the `data_location` is always set

Reply:
ok